src/modules/keyboardmanager/test/KeyboardManagerRemapLogicTest.cpp

```diff
@@ -0,0 +1,1311 @@
+            mockedInputHandler.SendVirtualInput(nInputs, input, sizeof(INPUT));
+
+            // LWin, RWin, A, Alt, V key states should be unchanged
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_LWIN), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_RWIN), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x41), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_MENU), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x56), false);
+
+            // Case 1.2
+            input[0].type = INPUT_KEYBOARD;
+            input[0].ki.wVk = VK_RWIN;
+            input[0].ki.dwFlags = 0;
+            input[1].type = INPUT_KEYBOARD;
+            input[1].ki.wVk = 0x41;
+            input[1].ki.dwFlags = 0;
+
+            // Send RWin+A keydown
+            mockedInputHandler.SendVirtualInput(nInputs, input, sizeof(INPUT));
+
+            // LWin, RWin, A key states should be unchanged, Alt, V should be true
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_LWIN), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_RWIN), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x41), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_MENU), true);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x56), true);
+
+            input[0].type = INPUT_KEYBOARD;
+            input[0].ki.wVk = 0x41;
+            input[0].ki.dwFlags = KEYEVENTF_KEYUP;
+            input[1].type = INPUT_KEYBOARD;
+            input[1].ki.wVk = VK_RWIN;
+            input[1].ki.dwFlags = KEYEVENTF_KEYUP;
+
+            // Release RWin+A
+            mockedInputHandler.SendVirtualInput(nInputs, input, sizeof(INPUT));
+
+            // LWin, RWin, A, Alt, V key states should be unchanged
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_LWIN), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_RWIN), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x41), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_MENU), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x56), false);
+
+            // Case 2
+            input[0].type = INPUT_KEYBOARD;
+            input[0].ki.wVk = VK_MENU;
+            input[0].ki.dwFlags = 0;
+            input[1].type = INPUT_KEYBOARD;
+            input[1].ki.wVk = 0x44;
+            input[1].ki.dwFlags = 0;
+
+            // Send Alt+D keydown
+            mockedInputHandler.SendVirtualInput(nInputs, input, sizeof(INPUT));
+
+            // Alt, D, RWin key states should be unchanged, LWin, B should be true
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_LWIN), true);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(VK_RWIN), false);
```

Comment:
Not directly related to this PR, but is there a reason it is equivalent to LWIN and not RWIN (Other than the fact that most keyboards don't have RWIN key)?

Reply:
Sounds good!